Solaris port fixes: sdksyms.sh tolerates cpp spacing in visibility
attributes (as in Oracle solaris-userland 30-solaris-port.patch), the
modesetting driver links the Xorg externs mapfile and libdrm, and
XYToWindow is exported for modules.

diff --git a/hw/xfree86/sdksyms.sh b/hw/xfree86/sdksyms.sh
index 5fd5b5c..bed6b7d 100755
--- a/hw/xfree86/sdksyms.sh
+++ b/hw/xfree86/sdksyms.sh
@@ -295,7 +295,13 @@ shift
 LC_ALL=C
 export LC_ALL
 ${CPP:-cpp} "$@" sdksyms.c > /dev/null || exit $?
-${CPP:-cpp} "$@" sdksyms.c | ${AWK:-awk} -v topdir=$topdir '
+# Some cpps add spaces in the middle of __attribute__((visibility(...)))
+# so we need to remove them for the following awk script to work.
+# Changes by alanc upstream
+${CPP:-cpp} "$@" sdksyms.c | \
+    ${SED:-sed} -e 's/( /(/g' -e 's/_ (/_(/g' -e 's/ )/)/g' \
+    -e 's/visibility (/visibility(/' | \
+    ${AWK:-awk} -v topdir=$topdir '
 function basename(file) {
     sub(".*/", "", file)
     return file
diff --git a/hw/xfree86/drivers/modesetting/Makefile.am b/hw/xfree86/drivers/modesetting/Makefile.am
index 961c574..99d9440 100644
--- a/hw/xfree86/drivers/modesetting/Makefile.am
+++ b/hw/xfree86/drivers/modesetting/Makefile.am
@@ -41,7 +41,8 @@ AM_CPPFLAGS = \
 
 modesetting_drv_la_LTLIBRARIES = modesetting_drv.la
 modesetting_drv_la_LDFLAGS = -module -avoid-version
-modesetting_drv_la_LIBADD = $(UDEV_LIBS) $(DRM_LIBS)
+modesetting_drv_la_LDFLAGS += $(XORG_EXTERNS_FLAG)
+modesetting_drv_la_LIBADD = $(UDEV_LIBS) $(DRM_LIBS) $(LIBDRM_LIBS)
 modesetting_drv_ladir = @moduledir@/drivers
 
 modesetting_drv_la_SOURCES = \
diff --git a/include/input.h b/include/input.h
index cdb5d5a..ea8daf2 100644
--- a/include/input.h
+++ b/include/input.h
@@ -685,7 +685,7 @@ void FixUpEventFromWindow(SpritePtr pSprite,
                           xEvent *xE,
                           WindowPtr pWin, Window child, Bool calcChild);
 extern Bool PointInBorderSize(WindowPtr pWin, int x, int y);
-extern WindowPtr XYToWindow(SpritePtr pSprite, int x, int y);
+extern _X_EXPORT WindowPtr XYToWindow(SpritePtr pSprite, int x, int y);
 extern int EventIsDeliverable(DeviceIntPtr dev, int evtype, WindowPtr win);
 extern Bool ActivatePassiveGrab(DeviceIntPtr dev, GrabPtr grab,
                                 InternalEvent *ev, InternalEvent *real_event);
